lib/Service/UserMigrationService.php

```diff
@@ -94,12 +100,71 @@ public function __construct(
 		$this->exportMapper = $exportMapper;
 		$this->importMapper = $importMapper;
 		$this->jobList = $jobList;
+		$this->internalCache = new CappedMemoryCache();
 
 		$this->mandatory = true;
 	}
 
 	/**
 	 * @param ?string[] $filteredMigratorList If not null, only these migrators will run. If empty only the main account data will be exported.
+	 *
+	 * @return int Estimated size in KiB
+	 */
+	private function estimateExportSize(IUser $user, ?array $filteredMigratorList = null): int {
+		$cacheKey = $user->getUID();
+		if ($filteredMigratorList !== null) {
+			$cacheKey .= '::' . json_encode($filteredMigratorList);
+		}
```

Comment:
It would be better to cache each migrator size on its own, so that adding a migrator to the list on the next call will only run the estimate for this one. And removing a migrator will only sum cache entries.